src/models/components/asset.ts

```diff
@@ -119,288 +148,1173 @@ export enum AssetPhase {
+};
 
-export class Hash extends SpeakeasyBase {
+export type Hash = {
     /**
      * Hash of the asset
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "hash" })
-    hash?: string;
-
+    hash?: string | undefined;
     /**
      * Hash algorithm used to compute the hash
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "algorithm" })
-    algorithm?: string;
-}
+    algorithm?: string | undefined;
+};
 
 /**
  * type of track
  */
-export enum AssetSchemasType {
+export enum AssetVideoSpecType {
     Video = "video",
     Audio = "audio",
 }
 
-export class Tracks extends SpeakeasyBase {
+export type Tracks = {
     /**
      * type of track
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "type" })
-    type: AssetSchemasType;
-
+    type: AssetVideoSpecType;
     /**
      * Codec of the track
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "codec" })
     codec: string;
-
     /**
      * Start time of the track in seconds
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "startTime" })
-    startTime?: number;
-
+    startTime?: number | undefined;
     /**
      * Duration of the track in seconds
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "duration" })
-    duration?: number;
-
+    duration?: number | undefined;
     /**
      * Bitrate of the track in bits per second
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "bitrate" })
-    bitrate?: number;
-
+    bitrate?: number | undefined;
     /**
      * Width of the track - only for video tracks
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "width" })
-    width?: number;
-
+    width?: number | undefined;
     /**
      * Height of the track - only for video tracks
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "height" })
-    height?: number;
-
+    height?: number | undefined;
     /**
      * Pixel format of the track - only for video tracks
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "pixelFormat" })
-    pixelFormat?: string;
-
+    pixelFormat?: string | undefined;
     /**
      * Frame rate of the track - only for video tracks
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "fps" })
-    fps?: number;
-
+    fps?: number | undefined;
     /**
      * Amount of audio channels in the track
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "channels" })
-    channels?: number;
-
+    channels?: number | undefined;
     /**
      * Sample rate of the track in samples per second - only for
      *
      * @remarks
      * audio tracks
      *
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "sampleRate" })
-    sampleRate?: number;
-
+    sampleRate?: number | undefined;
     /**
      * Bit depth of the track - only for audio tracks
      */
-    @SpeakeasyMetadata()
-    @Expose({ name: "bitDepth" })
-    bitDepth?: number;
-}
+    bitDepth?: number | undefined;
+};
 
 /**
  * Video metadata
  */
-export class VideoSpec extends SpeakeasyBase {
```

Comment:
I really like that this is not classes anymore - those were a pain to work with in RSC! This Zod schema generation is magical.